docs/CONFIG.md

```diff
@@ -6913,6 +7031,14 @@ block, but it is possible to receive a head BEFORE that block is actually
 available from the connected node via RPC, due to race conditions in the code of the remote ETH node. In this case you will get false
 "zero" blocks that are missing transactions.
 
+### FinalizedBlockOffset
+```toml
+FinalizedBlockOffset = 0 # Default
+```
+FinalizedBlockOffset defines the number of blocks by which the latest finalized block will be shifted. E.g if RPC returns block 100 as latest finalized, node will treat block `100 - FinalizedBlockOffset` as the latest finalized block.
+In case of `FinalityTagEnabled = false` Node will treat block `latest - FinalityDepth - FinalizedBlockOffset` as the latest finalized block.
+With `EnforceRepeatableRead = true`, RPC is considered healthy only if its most recent finalized block is larger or equal to the highest finalized block observed by the Node minus `FinalizedBlockOffset`.
```

Comment:
Wouldn't it be a bit cleaner & potentially safer if `FinalizedBlockOffset` is just ignored when `FinalityTagEnabled = false`, rather than adding it onto `FinalityDepth` to get an effectively deeper finality depth?

LogPoller still uses `FinalityDepth` for at least one thing, but it doesn't add `FinalizedBlockOffset` to that.  So now it seems that it will be using `FinalityDepth` for some things but `FinalityDepth + FinalizedBlockOffset` for other things (since it now relies on Headtracker to get the latest finalized block). Maybe this won't hurt anything, but seems awkward and error prone for future maintainability.

Reply:
@reductionista 
Hmm, one of the reasons why we need to use `FinalizedBlockOffset` is to avoid marking RPC as unhealthy only because it's slightly lagging behind. 
Example: RPC A's latest block is 100, RPC B - 98. If  `FinalizedBlockOffset=0` we can not route requests to RPC B as blocks that application layer considers finalised might change on RPC B. 
If we only drop `FinalizedBlockOffset` in HeadTracker's calculations, we'll have to make sure that the configured FinalityDepth accounts for `FinalizedBlockOffset,` in other words, `FinalityDepth = actual finality depth + FinalizedBlockOffset.`


In general, usage of the latest finalized block, which was not provided by HeadTracker, is unsafe (with or without finality tags). And we should not optimize for it. 
LogPoller's `fetchBlocks` is an exception:
-  it's self-contained. All the checks are performed on data received from the same RPC.
- It's performed on top of MultiNode's checks and not instead of them.

Reply:
Alright then! Just wanted to make sure you'd thought about this.